packages/openapi3/test/return-types.test.ts

```diff
@@ -199,6 +199,7 @@ describe("openapi3: return types", () => {
     ok(res.components.schemas.Error);
     deepStrictEqual(res.paths["/"].get.responses["default"].content["application/json"].schema, {
       $ref: "#/components/schemas/Error",
+      description: "Error",
```

Comment:
We do this in OpenAPI2, but I am not sure it is actually allowed in the OpenAPI schema

Reply:
In OpenAPI 2 and OpenAPI 3.0, the behavior of sibling properties of $ref is technically "undefined" -- any many tools like Spectral flag these as an error. We should avoid doing this if at all possible.

Reply:
@mikekistler I did this specifically because of this: https://gist.github.com/tjprescott/733cdb1723ecb2a7b791c8eac0cd5058#case-1a---extends-recordnamed-scalar-extends-unknown-with-doc

I could add some special case logic that erases any siblings of "$ref" but this then feels wrong and perhaps we should not pushing description into additionalProperties at all.

Reply:
Actually, right now I'm not able to craft a scenario where I can decorate the value type with `@doc` and _not_ have OpenAPI3 generate a `$ref` schema reference, so I'm not sure how to support this scenario.

Reply:
I removed the logic pushing the documentation into additional properties. The OpenAPI3 emitter always will generate a `$ref` anyways, even for custom scalars.